addons/advanced_ballistics/functions/fnc_handleFirePFH.sqf

```diff
@@ -29,9 +28,15 @@
         };
 
         _bullet setVelocity (_bulletVelocity vectorAdd (parseSimpleArray ("ace_advanced_ballistics" callExtension format["simulate:%1:%2:%3:%4:%5:%6", _index, _bulletVelocity, _bulletPosition, wind, ASLToATL(_bulletPosition) select 2, CBA_missionTime toFixed 6])));
+    } else {
+        GVAR(allBullets) set [_forEachIndex, objNull];
+        _deleted = true;
     };
-    nil
-} count +GVAR(allBullets);
+} forEach GVAR(allBullets);
+
+if (_deleted) then {
+    GVAR(allBullets) = GVAR(allBullets) - [objNull];
+};
 
 if (GVAR(allBullets) isEqualTo []) then {
```

Comment:
I think you could move 41-44 inside the `_deleted` if statement
(only way it could be `[]` is if there was a deletion)

Reply:
I changed it for now. but the question is if this is really required for "performance" or if we should add the PFH only once and never remove it because from my POV this PFH is only doing work while a bullet is in the array

Reply:
I think this would be good, windDeflection pfeh works that way